homeassistant/components/homekit/__init__.py

```diff
@@ -118,19 +118,19 @@ def get_accessory(hass, state, aid, config):
         unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
         device_class = state.attributes.get(ATTR_DEVICE_CLASS)
 
-        if device_class == DEVICE_CLASS_TEMPERATURE or unit == TEMP_CELSIUS \
-                or unit == TEMP_FAHRENHEIT:
+        if device_class == DEVICE_CLASS_TEMPERATURE and (
+                unit == TEMP_CELSIUS or unit == TEMP_FAHRENHEIT):
             a_type = 'TemperatureSensor'
-        elif device_class == DEVICE_CLASS_HUMIDITY or unit == '%':
+        elif device_class == DEVICE_CLASS_HUMIDITY and unit == '%':
             a_type = 'HumiditySensor'
         elif device_class == DEVICE_CLASS_PM25 \
                 or DEVICE_CLASS_PM25 in state.entity_id:
             a_type = 'AirQualitySensor'
         elif device_class == DEVICE_CLASS_CO2 \
                 or DEVICE_CLASS_CO2 in state.entity_id:
             a_type = 'CarbonDioxideSensor'
-        elif device_class == DEVICE_CLASS_LIGHT or unit == 'lm' or \
-                unit == 'lux' or unit == 'lx':
+        elif device_class == DEVICE_CLASS_LIGHT and (
+                unit == 'lm' or unit == 'lux' or unit == 'lx'):
```

Comment:
There's no such device class "light" officially supported: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/__init__.py

```python
unit in ('lm', 'lux', 'lx')
```

Reply:
The `device_class = 'light'` was taken from the `binary_sensor`, but I agree it should be official. I will create a PR for it shortly.

Btw. I like the solution with `unit in ('lm', 'lux', 'lx')`. Didn't though of that.